### PR TITLE
Add wb-suggest to the search field

### PR DESCRIPTION
**Pull request by ipaksc**
Related to WET-213

### CODE REVIEW

File: docs/static-header-footer/bootstrap-3.html [hunk 1]
```diff
@@ -42,6 +42,9 @@ <h2>Search</h2>
 						<label for="wb-srch-q" class="wb-inv">Search Canada.ca</label>
 						<input id="wb-srch-q" list="wb-srch-q-ac" class="wb-srch-q form-control" name="q" type="search" value="" size="34" maxlength="170" placeholder="Search Canada.ca">
 						<datalist id="wb-srch-q-ac">
+							data-wb-limit="10"
+							data-wb-filter-type="startWith"
+							data-wb-suggest="https://wet-boew.github.io/themes-dist/GCWeb/GCWeb/sites/search/json/canada-en.json#/suggestions">
```
Comment (duboisp) [inline docs/static-header-footer/bootstrap-3.html:47]: Don't need to specify the domain "wet-boew.github.io".

File: sites/headers-includes/sitesearch.html [hunk 2]
```diff
@@ -5,7 +5,11 @@ <h2>{{ i18nText-search }}</h2>
 		<div class="form-group wb-srch-qry">
 			<label for="wb-srch-q" class="wb-inv">{{ i18nText-searchSite }}</label>
 			<input id="wb-srch-q" list="wb-srch-q-ac" class="wb-srch-q form-control" name="q" type="search" value="" size="34" maxlength="170" placeholder="{{ i18nText-searchSite }}" />
-			<datalist id="wb-srch-q-ac"></datalist>
+			<datalist id="wb-srch-q-ac">
+				data-wb-limit="10"
+				data-wb-filter-type="startWith"
```
Comment (duboisp) [inline sites/headers-includes/sitesearch.html:10]:
```suggestion
```

File: docs/static-header-footer/bootstrap-3.html [hunk 3]
```diff
@@ -42,6 +42,9 @@ <h2>Search</h2>
 						<label for="wb-srch-q" class="wb-inv">Search Canada.ca</label>
 						<input id="wb-srch-q" list="wb-srch-q-ac" class="wb-srch-q form-control" name="q" type="search" value="" size="34" maxlength="170" placeholder="Search Canada.ca">
 						<datalist id="wb-srch-q-ac">
+							data-wb-limit="10"
+							data-wb-filter-type="startWith"
```
Comment (duboisp) [inline docs/static-header-footer/bootstrap-3.html:46]: Why do we need does configuration?
```suggestion
```

File: sites/headers-includes/sitesearch.html [hunk 4]
```diff
@@ -5,7 +5,11 @@ <h2>{{ i18nText-search }}</h2>
...
+				data-wb-suggest="https://wet-boew.github.io/themes-dist/GCWeb/GCWeb/sites/search/json/canada-{{ i18nText-lang }}.json#/suggestions">
```
Comment (duboisp) [inline sites/headers-includes/sitesearch.html:11]: Don't need to specify the domain "wet-boew.github.io".

File: sites/search/json/canada-en.json [hunk 5]
```diff
@@ -0,0 +1,487 @@
+{
+  "@context": {
+    "dc": "http://purl.org/dc/terms/",
+    "suggestions": {
+      "@id": "http://www.w3.org/2001/XMLSchema#string",
+      "@container": "@list"
+    }
+  },
+  "@language": "en",
+  "dc:title": "Global top search term",
```
Comment (duboisp) [inline sites/search/json/canada-en.json:10]: I suggest to be more specific in the title and clearer. 
That title raise me the following question:
* "Global" to what? 
* What is the difference between "Global" and "top"? Is it not the same?

File: sites/search/json/canada-en.json [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+  "dc:modified": "2018-02-06",
```
Comment (duboisp) [inline sites/search/json/canada-en.json:11]: Is that the real date we have reviewed the top search term list? If so, it is outdated for sure.

File: sites/search/json/canada-fr.json [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+  "@context": {
+    "dc": "http://purl.org/dc/terms/",
+    "suggestions": {
+      "@id": "http://www.w3.org/2001/XMLSchema#string",
+      "@container": "@list"
+    }
+  },
+  "@language": "fr",
+  "dc:title": "Terme de recherche globaux",
+  "dc:modified": "2018-02-06",
```
Comment (duboisp) [inline sites/search/json/canada-fr.json:11]: Idem as commented in English.

**Review by duboisp — CHANGES_REQUESTED** [review]

File: sites/search/json/canada-en.json (hunk 6 above)
Comment (ipaksc) [reply sites/search/json/canada-en.json:11]: I downloaded the files from AEM. I am not sure if our team is responsible for updating the content of the document.

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html:46]: I copied the configuration from canada.ca.

<datalist id="datalist-sch-inp-ac" 
			data-wb-limit="10" 
			data-wb-filter-type="startWith" 
			data-wb-suggest="/content/dam/canada/srch-rchr/canada-en.json#/suggestions"> 
	</datalist>

Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html:46]: `<datalist id="datalist-sch-inp-ac" 
			data-wb-limit="10" 
			data-wb-filter-type="startWith" 
			data-wb-suggest="/content/dam/canada/srch-rchr/canada-en.json#/suggestions"> 
	</datalist>`

File: sites/search/json/canada-en.json (hunk 6 above)
Comment (ipaksc) [reply sites/search/json/canada-en.json:11]: I have asked the analytic team to provide a list of most popular search terms.

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (duboisp) [reply docs/static-header-footer/bootstrap-3.html:46]: Sorry, that is not a rational.

File: sites/search/json/canada-en.json (hunk 6 above)
Comment (duboisp) [reply sites/search/json/canada-en.json:11]: You will need to find out the process of who and how that file is updated. And then ensure the person that are making the update are also updating the modified date. And let that person know to update/review the title too.
Is it updated by analytic team??? They should not update/manage any content on AEM, usually that is done by our publisher or in rare situation by the Dev team

**Comment by duboisp** [comment]
We might need to do some additional change, like 
* creating a working example + documentation page in /sites/search folder.
* creating an index.json-ld file to hold the top right search component
* Extract the top right search component from the header and move it into the /sites/search folder

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html:46]: What would be the right way to do it?

**Comment by coprisanu** [comment]
> We might need to do some additional change, like
> 
>     * creating a working example + documentation page in /sites/search folder.
> 
>     * creating an index.json-ld file to hold the top right search component
> 
>     * Extract the top right search component from the header and move it into the /sites/search folder



> We might need to do some additional change, like
> 
>     * creating a working example + documentation page in /sites/search folder.
> 
>     * creating an index.json-ld file to hold the top right search component
> 
>     * Extract the top right search component from the header and move it into the /sites/search folder

@duboisp I wanted to understand better your suggested changes #2 (creating an index.json-ld file to hold the top right search component) and #3 (Extract the top right search component from the header and move it into the /sites/search folder). Are you suggesting only to change the structure in github? We do need the top right search box in the header as it is inherited by all the other pages.

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (duboisp) [reply docs/static-header-footer/bootstrap-3.html:46]: Use it as per it's default configuration like:
```html
<datalist id="datalist-sch-inp-ac" data-wb-suggest="//www.canada.ca/content/dam/canada/srch-rchr/canada-en.json#/suggestions"> </datalist>
```

hum looking at the `id` of the `<datalist>` element, we might need to review it. It should be prefixed, like with "gc-" something. @GormFrank can you provide a suggestion for the `id` value?

**Comment by duboisp** [comment]
@coprisanu all the 3 bullets from my last comments :

> * creating a working example + documentation page in /sites/search folder.
> * creating an index.json-ld file to hold the top right search component
> * Extract the top right search component from the header and move it into the /sites/search folder

are related to getting that feature well documented in GCWeb. There is a big documentation gap where a lot of components are missing technical documentation. And this top right search is one of them. Although, don't take this as a blocker of this PR but instead something that we need to consider to get it fixed at our earliest convenience (which must not be calculated in years :wink:).

File: sites/search/json/canada-en.json (hunk 5 above)
Comment (ipaksc) [reply sites/search/json/canada-en.json:10]: Christian suggests this title "List of suggestions for the autocomplete function in the top-right search box" if you are agree we will send it for translation. cc @coprisanu

Comment (duboisp) [reply sites/search/json/canada-en.json:10]: @ipaksc do you know the current process on how that file is updated?

That propose name is long. Can we strip out "for the autocomplete function"? Include a qualifier this is for a sites level components. Use the current component name with is "search" and not "top-right search box".
Something like: "List of suggestions for the sites template search component"

\cc @coprisanu @GormFrank

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (GormFrank) [reply docs/static-header-footer/bootstrap-3.html:46]: @ipaksc @duboisp I think the `id` should be something like: `gc-srch-q-list`

Comment (duboisp) [reply docs/static-header-footer/bootstrap-3.html:46]: I didn't saw we already had an id for the datalist, use this instead `wb-srch-q-ac`

**Review by duboisp — CHANGES_REQUESTED** [review]

File: docs/static-header-footer/bootstrap-3.html (hunk 1 above)
Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html]: Changed to data-wb-suggest="https://www.canada.ca/content/dam/canada/srch-rchr/canada-en.json#/suggestions"

File: sites/headers-includes/sitesearch.html (hunk 2 above)
Comment (ipaksc) [reply sites/headers-includes/sitesearch.html]: Removed

File: docs/static-header-footer/bootstrap-3.html (hunk 3 above)
Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html]: Removed

File: sites/headers-includes/sitesearch.html (hunk 4 above)
Comment (ipaksc) [reply sites/headers-includes/sitesearch.html]: Changed to data-wb-suggest="https://www.canada.ca/content/dam/canada/srch-rchr/canada-{{ i18nText-lang }}.json#/suggestions

File: docs/static-header-footer/bootstrap-3.html [hunk 8]
```diff
@@ -42,6 +42,7 @@ <h2>Search</h2>
 						<label for="wb-srch-q" class="wb-inv">Search Canada.ca</label>
 						<input id="wb-srch-q" list="wb-srch-q-ac" class="wb-srch-q form-control" name="q" type="search" value="" size="34" maxlength="170" placeholder="Search Canada.ca">
 						<datalist id="wb-srch-q-ac">
```
Comment (GormFrank) [inline docs/static-header-footer/bootstrap-3.html:44]:
```suggestion
						<datalist id="wb-srch-q-ac"
```

File: sites/headers-includes/sitesearch.html [hunk 9]
```diff
@@ -5,7 +5,9 @@ <h2>{{ i18nText-search }}</h2>
 		<div class="form-group wb-srch-qry">
 			<label for="wb-srch-q" class="wb-inv">{{ i18nText-searchSite }}</label>
 			<input id="wb-srch-q" list="wb-srch-q-ac" class="wb-srch-q form-control" name="q" type="search" value="" size="34" maxlength="170" placeholder="{{ i18nText-searchSite }}" />
-			<datalist id="wb-srch-q-ac"></datalist>
+			<datalist id="wb-srch-q-ac">
```
Comment (GormFrank) [inline sites/headers-includes/sitesearch.html:8]:
```suggestion
			<datalist id="wb-srch-q-ac"
```

**Review by GormFrank — CHANGES_REQUESTED** [review]

File: docs/static-header-footer/bootstrap-3.html (hunk 8 above)
Comment (ipaksc) [reply docs/static-header-footer/bootstrap-3.html]: @GormFrank It's fixed

File: sites/headers-includes/sitesearch.html (hunk 9 above)
Comment (ipaksc) [reply sites/headers-includes/sitesearch.html]: @GormFrank It's fixed

**Review by duboisp — CHANGES_REQUESTED** [review]
We will need a review process to ensure the top search term are up to date and remain accurate.

**Pull request closed without merging**